src/main/resources/edu/cornell/kfs/module/cg/cu-ojb-cg.xml

```diff
@@ -125,6 +251,92 @@
+        <field-descriptor name="lastUpdatedTimestamp" column="LAST_UPDT_TS" jdbc-type="TIMESTAMP" index="true"/>
+
+        <field-descriptor name="proposalDirectCostAmount" column="CGPRPSL_DRCTCS_AMT" jdbc-type="DECIMAL"
+                          conversion="org.kuali.kfs.core.framework.persistence.ojb.conversion.OjbKualiDecimalFieldConversion"/>
+        <field-descriptor name="proposalIndirectCostAmount" column="CGPRPSL_INDRCS_AMT" jdbc-type="DECIMAL"
+                          conversion="org.kuali.kfs.core.framework.persistence.ojb.conversion.OjbKualiDecimalFieldConversion"/>
+        <field-descriptor name="proposalRejectedDate" column="CGPRPSL_REJECTD_DT" jdbc-type="DATE"/>
+        <field-descriptor name="proposalLastUpdateDate" column="CGPRPSL_LSTUPDT_DT" jdbc-type="TIMESTAMP"/>
+        <field-descriptor name="proposalDueDate" column="CGPRPSL_DUE_DT" jdbc-type="DATE"/>
+        <field-descriptor name="proposalTotalProjectAmount" column="CGPRPSL_TOTPRJ_AMT" jdbc-type="DECIMAL"
+                          conversion="org.kuali.kfs.core.framework.persistence.ojb.conversion.OjbKualiDecimalFieldConversion"/>
+        <field-descriptor name="proposalSubmissionDate" column="CGPRPSL_SUBMSSN_DT" jdbc-type="DATE"/>
+        <field-descriptor name="proposalFederalPassThroughIndicator" column="CGPRPSL_FEDPT_IND" jdbc-type="VARCHAR"
+                          conversion="org.kuali.kfs.core.framework.persistence.ojb.conversion.OjbCharBooleanConversion"/>
+        <field-descriptor name="oldProposalNumber" column="CG_OLD_PRPSL_NBR" jdbc-type="VARCHAR"/>
+        <field-descriptor name="grantNumber" column="CG_GRANT_NBR" jdbc-type="VARCHAR"/>
+        <field-descriptor name="proposalClosingDate" column="CGPRPSL_CLOSING_DT" jdbc-type="DATE"/>
+        <field-descriptor name="proposalAwardTypeCode" column="CGPRPSL_AWD_TYP_CD" jdbc-type="VARCHAR" index="true"/>
+        <field-descriptor name="agencyNumber" column="CG_AGENCY_NBR" jdbc-type="VARCHAR" index="true"/>
+        <field-descriptor name="proposalStatusCode" column="CGPRPSL_STAT_CD" jdbc-type="VARCHAR" index="true"/>
+        <field-descriptor name="federalPassThroughAgencyNumber" column="CG_FEDPT_AGNCY_NBR" jdbc-type="VARCHAR" index="true"/>
+        <field-descriptor name="cfdaNumber" column="CG_CFDA_NBR" jdbc-type="VARCHAR"/>
+        <field-descriptor name="proposalFellowName" column="CGPRPSL_FELLOW_NM" jdbc-type="VARCHAR"/>
+        <field-descriptor name="proposalPurposeCode" column="CGPRPSL_PURPOSE_CD" jdbc-type="VARCHAR" index="true"/>
+        <field-descriptor name="proposalProjectTitle" column="CGPRPSL_PROJ_TTL" jdbc-type="VARCHAR"/>
+        <field-descriptor name="active" column="ROW_ACTV_IND" jdbc-type="VARCHAR" conversion="org.kuali.kfs.core.framework.persistence.ojb.conversion.OjbCharBooleanConversion"/>
+        <reference-descriptor name="award" class-ref="edu.cornell.kfs.module.cg.businessobject.CuAward" auto-retrieve="true" auto-update="none"
```

Comment:
I think you mentioned in the standup this OJB mapping is required due a previously fixed problem.  Should that previous fix be re-engineered to avoid needing this mapping (if so a different user story may be needed for that)?  I am concerned with are going to need update more and more object mapping in OJB just to link to CUAward.  Is there away to avoid this?  Am I overly concerned?

Reply:
The only objects that I found that have OJB reference/collection mappings to Award were AwardAccount and Proposal. The other references should already be covered by our existing EBO mapping customizations and the new row-mapping customizations from this PR.

The alternative would be to remove the CuAward BO and transfer its custom column data to the extended attributes, and also move our custom "populateFromProposal" method handling to a separate helper class. The latter point is the biggest concern since that could increase the risk of accidentally overlooking changes to that method when generating upgrade reports, unless we decide to overlay Award to customize that method.

Replacing CuAward with just the base Award would require more work in the short term, but perhaps it would provide more of a benefit in the long term if we can be sure not to overlook future KualiCo changes to the "populateFromProposal" method (and also transfer the relevant data to the extended attribute instead). I'll start looking into this approach a little more to see just how much work it would be.